R/gen_population.R

```diff
@@ -6,64 +6,131 @@
 #' @param total_pop The overal population count of interest - the default is
 #'   1000 people
 #' @param groups A character vector of groups - the default is set for age
-#' groups: c("0-4","5-14","15-24","25-34","35-44","45+")
+#' groups: c("0-4","5-14","15-29","30-44","45+")
 #' @param strata A character vector for stratifying groups -
 #'   the default is set for gender: c("Male", "Female")
 #' @param proportions A numeric vector specifying the proportions (as decimals)
-#'   for each age group - the default repeats c(0.182, 0.278, 0.26, 0.11, 0.07,
-#'   0.10) for genders.  However you can change this manually, make sure to
-#'   have the length equal to groups times strata.
+#'   for each group - the default repeats c(0.079, 0.134, 0.139, 0.082, 0.067)
+#'   for genders.  However you can change this manually, make sure to
+#'   have the length equal to groups times strata (or half thereof).
+#'   These defaults are based of MSF general emergency intervention standard
+#'   values.
+#' @param counts A numeric vector specifying the counts for each group.
+#'   The default is NULL - as most often proportions above will be used.
+#'   If is not NULL then total_pop and proportions will be ignored.
+#'   Make sure the length of this vector is equal to groups time strata (or if it
+#'   is half then it will repeat for each strata).
+#'   For reference, the MSF general emergency intervention standard values
+#'   are c(7945, 13391, 13861, 8138, 6665) based on above
+#'   groups for a 100,000 person population.
 #' @param tibble Return data as a tidyverse tibble (default is TRUE)
 #' @importFrom dplyr bind_cols
 #' @export
 
 
 gen_population <- function(total_pop = 1000,
-                           groups = c("0-4","5-14","15-24","25-34","35-44","45+"),
+                           groups = c("0-4","5-14","15-29","30-44","45+"),
                            strata = c("Male", "Female"),
-                           proportions = c(0.182, 0.278, 0.26, 0.11, 0.07, 0.10),
+                           proportions = c(0.079, 0.134, 0.139, 0.082, 0.066),
+                           counts = NULL,
                            tibble = TRUE) {
 
-  if (length(proportions) != length(groups) & is.null(strata)) {
-    difference <- abs(length(groups) - length(proportions))
+  # using proportions (where counts not specified)
+  if (is.null(counts)) {
 
-    warning(sprintf("Proporitons and groups do not match, without specifying strata.
-                    The difference in length was %d (proportions may have been dropped)",
-                    difference))
+    if (length(proportions) != length(groups) & is.null(strata)) {
+      difference <- abs(length(groups) - length(proportions))
 
-    proportions <- proportions[1:length(groups)]
-  }
+      stop(sprintf("Proporitons and groups do not match, without specifying strata.
```

Comment:
when you construct stop messages like these, the whitespace becomes part of the message. It would be better to construct the message with `glue`:

``` r
differences <- 69
stop(glue::glue("Proportions and groups do not match\n", 
                "The difference in length was {differences}"))
#> Error in eval(expr, envir, enclos): Proportions and groups do not match
#> The difference in length was 69
```

<sup>Created on 2019-08-20 by the [reprex package](https://reprex.tidyverse.org) (v0.3.0)</sup>